<?xml version="1.0"?>
<launch>

    <!-- Launch file arguments -->
    <arg name="map"             default="sml"/>
    <arg name="is_sim"          default="true"/>
    <arg name="use_rviz"        default="true"/>
    <arg name="start_rviz"      default="false"/>
    <arg name="initial_pose_x"  default="-2.65488696"/>
    <arg name="initial_pose_y"  default="-1.64422277"/>
    <arg name="initial_pose_a"  default="1.57" /> <!-- wrt to map-->

    <!-- Start map server -->
    <node name="map_server" pkg="map_server" type="map_server" args="$(find svea_core)/maps/$(arg map).yaml" output="screen"/>

    <!-- If is_sim equal to false, then include all these tags-->
    <group unless="$(arg is_sim)">

        <!-- Start low-level interface -->
        <node name="serial_node" pkg="rosserial_python" type="serial_node.py">
            <param name="port" value="/dev/ttyACM0"/>
            <param name="baud" value="250000"/>
        </node>

        <!-- Start localization -->
        <include file="$(find svea_sensors)/launch/localize.launch">
            <arg name="initial_pose_x" value="$(arg initial_pose_x)" />
            <arg name="initial_pose_y" value="$(arg initial_pose_y)" />
            <arg name="initial_pose_a" value="$(arg initial_pose_a)" />
        </include>
    </group>

    <!-- Start RViz -->
    <node if="$(eval use_rviz and start_rviz)" name="rviz" pkg="rviz" type="rviz" args="-d $(find svea_core)/rviz/SVEA_floor2.rviz"/>
    <!-- Start pure_pursuit -->
    <node name="draft" pkg="thesis_mvp" type="draft.py" output="screen">
        <param name="use_rviz" value="$(arg use_rviz)"/>
        <param name="is_sim" value="$(arg is_sim)"/>
        <!-- Waypoints for floo2 map -->
        <rosparam>
            track_start: [2.5, -1, 1.54]
            # Big Track
            arcs:
            - [2.5]
            - [0.5, 90]
            - [4.0]
            - [0.5, 90]
            - [3.5]
            - [0.5, 90]
            - [4.0]
            - [0.5, 90]
            - [1.0]
            # Small Track
            - [0.5, 90]
            - [4.0]
            - [0.5, 90]
            - [1.0]
            - [0.5, 90]
            - [4.0]
            - [0.5, 90]
            - [1.0]
        </rosparam>
    </node>
</launch>
